Use institution name in email signature for #3380

sylvieed opened the pull request:
_Resolves #3380_

This adds a setting for collections to use the collection owner's institution name in user mailer email signatures instead of "FromThePage". It adds a new boolean field to the `collection` model, `institution_signature`. It's defaulted to `false`, which means user emails will be signed with FromThePage. If it's updates to `true`, user emails will be signed with the owner's display name.
I chose to keep this setting simple and just use a toggle instead of a fully customizable signature with a textbox because I assumed institutions wouldn't want to sign emails with anything other than their set display name.

Here's the setting:

![setting](https://user-images.githubusercontent.com/35716893/207419012-05491946-19b9-4715-8bd6-b4d574be8039.jpg)
("Drexel Libraries" Is whatever the collection owner's display name is)

Here's how the emails look, with the customized signature:
<img alt="html email" src="https://user-images.githubusercontent.com/35716893/207417072-dfb75443-ffd5-4703-bb80-dd193e6584a9.jpg" width="70%">

and default signature: (this is unchanged)
<img alt="html email fromthepage" src="https://user-images.githubusercontent.com/35716893/207417075-98906125-e2d0-4129-8bf9-32d76631906f.jpg" width="70%">

The emails this updates:
* Note added
* Collection collaborator
* Collection reviewer
* Work collaborator

The other user mailer emails are unchanged because they don't have a signature. The nightly user activity email is unchanged because it's not from a specific institution. Both text and html emails are updated.

saracarl commented:
Would we want the institution name in quotes?  

This implies that responses go to the institution; they don't.  Should we set a reply-to that's the institution's account email?  (Complicated issues)  Maybe just a disclaimer or a "Please respond on the page here rather than responding to this email"

sylvieed commented:
Added quotes around the institution names in the setting description:

![quotes](https://user-images.githubusercontent.com/35716893/207900578-444857bd-e7be-4c1d-be11-49794fda366d.jpg)

sylvieed commented:
Changed the note added email to say "respond to the note" instead of "view the note" so it's more clear where the users should reply.

![respond to](https://user-images.githubusercontent.com/35716893/207929415-3a9f76bc-d9cb-470d-8407-171873b13fd2.jpg)

Added a reply-to to the emails from a specific collection to the collection owner's email. The reply-to is always there, not only when the owners check the "institution signature" checkbox for that collection.

![replyto](https://user-images.githubusercontent.com/35716893/207929769-4eb8cadf-3c11-4cd8-aef5-864dfa97af17.jpg)

saracarl commented on db/migrate/20221213173527_add_institution_signature_to_collection.rb:
```diff
@@ -0,0 +1,5 @@
+class AddInstitutionSignatureToCollection < ActiveRecord::Migration[6.0]
+  def change
+    add_column :collections, :institution_signature, :boolean, default: false
```
Default to true.

sylvieed replied:
Could we just remove the `institution_signature` column completely because it won't be used anywhere?

saracarl reviewed (changes requested):
Hi Sylvie,

We talked about this some more, and realized that this option should be what happens all the time.  In other words, we shouldn't give institutions the option to choose.

So....  could you please back out the UI to choose sending with the institution name, and instead default to true.  You won't need the schema change because it will always be true.

Sometimes the institution's display name is blank -- we're not sure why (some sort of editing the profile causes this problem, if you want to look into that -- or search for related issues we might have one.).  Let's handle that case either by figuring it out, or more likely checking for a blank display name and using FromThePage in that case.

sylvieed commented:
I removed the `institution_signature` column as it won't be used anymore.

Now, all of the user mailer emails that are from a specific project owner (i.e. about a specific collection, work) are signed by that project owner unless they have a blank display name, and then it's signed by FromThePage.

saracarl commented on db/migrate/20221219210301_remove_institution_signature_from_collection.rb:
```diff
@@ -0,0 +1,5 @@
+class RemoveInstitutionSignatureFromCollection < ActiveRecord::Migration[6.0]
+  def change
+    remove_column :collections, :institution_signature, :boolean
```
So in a more perfectionist environment, you'd probably remove the add migration for this column instead of adding the column and then removing it.  But we're going to roll with it because we don't care that much.

sylvieed replied:
Makes sense, thanks
